### PR TITLE
updated stretch to factor x and y columns by default

### PR DESCRIPTION
**Pull request by mattwarkentin**

### CODE REVIEW

**Comment by mattwarkentin** [comment]
Fixes #98

**Comment by mattwarkentin** [comment]
These checks did not fail when I ran R-CMD-check locally. Maybe my version of `dplyr::glimpse()` is out-of-date?

**Comment by juliasilge** [comment]
Closes #79, #84, and #98

**Comment by mattwarkentin** [comment]
Hi @juliasilge,

It looks like you might be preparing to review this PR. Please let me know if there is anything that isn't clear or if there is anything else I can do to help.

**Comment by juliasilge** [comment]
Thank you so much @mattwarkentin! This is looking great. 🙌  We are going to have @thisisdaryn do a little review here in a bit with a few changes.

**Comment by mattwarkentin** [comment]
No problem! Happy to help.

File: R/cor_df.R [hunk 1]
```diff
@@ -123,7 +124,7 @@ rplot.cor_df <- function(rdf,
   row_order <- rdf$rowname
 
   # Convert data to relevant format for plotting
-  pd <- stretch(rdf, na.rm = TRUE)
+  pd <- stretch(rdf, na.rm = TRUE, .order = .order)
```
Comment (thisisdaryn) [inline R/cor_df.R:127]: `stretch` does not need to be modified necessarily. (I'm adding another comment to the proposed changes in the body of the function). 

The call to `stretch` itself seems to be fine as it preserves the order in both x and y columns as is. It is only necessary to use factors internal to `rplot` to get `ggplot` to override its default alphabetical ordering. 

The following lines between the calls to `stretch` and `ggplot` within `rplot` will get the plot to reflect the order of the variables in the correlation matrix. 

```
  if(.order != "alphabet"){
    if(.order == "default"){
      pd$x <- factor(pd$x, levels = row_order)
      pd$y <- factor(pd$y, levels = rev(row_order))
    }
  }
```
I've structured the above branching logic to accommodate possible future non-alphabetical choices of ordering.

Someone using `ggplot` manually with a data.frame output from stretch would need to replicate the factor conversion. Which is standard. My preference is for type stability/consistency of `stretch`. 

I think changing the factors internally in `rplot` is appropriate as it is a plot meant specifically to represent a correlation matrix as the matrix has been received.

File: R/reshape.R [hunk 2]
```diff
@@ -116,16 +119,23 @@ focus_if.default <- function(x, .predicate, ..., mirror = FALSE) {
...
 #' @export
-stretch.cor_df <- function(x, na.rm = FALSE, remove.dups =  FALSE) {
+stretch.cor_df <- function(x, na.rm = FALSE, remove.dups =  FALSE,
+                           .order = c("default", "alphabet")) {
+  .order <- match.arg(.order)
   if(remove.dups) x <- shave(x)
   row_name <- x$rowname
   x <- x[, colnames(x) != "rowname"]
   tb <- imap_dfr(x, ~tibble(x = .y, y = row_name, r = .x))
+  if(.order == "default")  {
```
Comment (thisisdaryn) [inline R/reshape.R:136]: These changes do not seem necessary to address the original issue. Rather, they would represent an expansion of the capabilities of `stretch`. If `stretch` is to be augmented to optionally sort in alphabetical order, another option may be to allow an alphabetical option to  `rearrange`.

File: R/cor_df.R (hunk 1 above)
Comment (juliasilge) [reply R/cor_df.R:127]: Ah, I got what you were saying now. 👍

**Pull request merged**